packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -131,7 +132,11 @@ export function createReactExtension<
         return (
           <Suspense fallback={<Progress />}>
             <PluginErrorBoundary app={app} plugin={plugin}>
-              <Component {...props} />
+              <ExtensionAwareContext.Provider
```

Comment:
Hm... Was thinking about the name and scope of this context. First, the "Aware" word caught my eye and it sounded like something you'd put on something that's stateful or enriches some other interface - rather than a static context that's not aware of much at all :)

So that led me to think, were you intending that this be a general context, not just for use by analytics? In that case perhaps just `ExtensionContext`?

But maybe, could it be valuable with a specific `AnalyticsContext`? Just a thought. What if there was a single `AnalyticsContext` at the root, and then you could at various points in the React tree have `AnalyticsContext.Define labels={{ 'plugin-id': x,...` and those were nestable and chained up the tree so that an emit actually sends the union of the labels all the way up to the root?

The reason I bring it up is, there will be a fair bit of nesting of plugins and widgets within pages etc, and maybe there is value for them to join forces?

Bikeshedding warning here :)

Reply:
Yeah, this is an interesting thought.  And in fact, as I was hacking on this, I had originally named this context `AnalyticsContext`!  But renamed at the end because...  It didn't have much to do with analytics, only really ending up providing some metadata about an extension boundary.

I was curious to see how contexts with self-inheritance would _feel_ and it's interesting.  Feels kinda fragile, but mostly works ([see example](https://codesandbox.io/s/react-context-inheritance-i06nw?file=/src/App.js:1169-1221)).

Implementation has this downside:

```tsx
const SubComponent = () => {
  const { captureEvent } = useAnalytics();
  return (
    <>
      <Analytics value={{ layer3: "Bottom-Level" }}>
        <button onClick={() => captureEvent({ from: "SubComponent" })}>
          Sub-Component
        </button>
      </Analytics>
    </>
  );
};
```

^ Instinct says that because `captureEvent` is nested under `<Analytics />`, you'd also get `layer3` passed as an attribute.  ...But you don't, because `captureEvent` is bound to the parent context.

----

It's an interesting pattern, but opening up the possibility for plugin developers to write to arbitrary attributes feels very _wild-west_ (nothing stopping you from overwriting a parent attribute in your sub-tree).

Reply:
Aight! Yeah I'll stop here and let the others review, had some thoughts but don't want to pollute the discussion too much while away and not very responsive :)

Reply:
Decided I liked this idea and implemented it!  Chose to call it `<AnalyticsDomain />`.

I also worked around my downside via a `withAnalyticsDomain` HOC provider, like this:

```ts
const SubComponent = () => {
  const { captureEvent } = useAnalytics();
  return (
    <button onClick={() => captureEvent({ from: "SubComponent" })}>
      Sub-Component
    </button>
  );
};

export default withAnalyticsDomain(SubComponent, { layer3: 'Bottom-Level' });
```